test/ecs/raycastHelperSystem.spec.ts

```diff
@@ -0,0 +1,412 @@
+import {
+  ColliderLayer,
+  components,
+  createRaycastSystem,
+  Engine,
+  IEngine,
+  RaycastSystem,
+  RaycastQueryType
+} from '../../packages/@dcl/ecs/src'
+import { Vector3 } from '../../packages/@dcl/sdk/math'
+
+describe('Raycast Helper System should', () => {
+  const engine: IEngine = Engine()
+  const raycastHelperSystem: RaycastSystem = createRaycastSystem(engine)
+  const raycastComponent = components.Raycast(engine)
+  const raycastResultComponent = components.RaycastResult(engine)
+
+  it('run callback on raycast result for LocalDirection', async () => {
+    const raycastEntity = engine.addEntity()
+    let callbacksCounter = 0
+    raycastHelperSystem.registerLocalDirectionRaycast(
+      raycastEntity,
+      (result) => {
+        callbacksCounter++
+      },
+      {
+        direction: Vector3.Forward(),
+        queryType: RaycastQueryType.RQT_HIT_FIRST
+      }
+    )
+
+    // Simulate client-side result attachment
+    raycastResultComponent.create(raycastEntity, {
+      hits: [],
+      direction: Vector3.Zero(),
+      globalOrigin: Vector3.Zero()
+    })
+
+    await engine.update(1)
+    expect(callbacksCounter).toBe(1)
+  })
+
+  it('run callback on raycast result for GlobalDirection', async () => {
+    const raycastEntity = engine.addEntity()
+    let callbacksCounter = 0
+    raycastHelperSystem.registerGlobalDirectionRaycast(
+      raycastEntity,
+      (result) => {
+        callbacksCounter++
+      },
+      {
+        direction: Vector3.Forward(),
+        queryType: RaycastQueryType.RQT_HIT_FIRST
+      }
+    )
+
+    // Simulate client-side result attachment
+    raycastResultComponent.create(raycastEntity, {
+      hits: [],
+      direction: Vector3.Zero(),
+      globalOrigin: Vector3.Zero()
+    })
+
+    await engine.update(1)
+    expect(callbacksCounter).toBe(1)
```

Comment:
I'd recommend to use jest.fn for this test

```suggestion
    let fn = jest.fn()
    raycastHelperSystem.registerGlobalDirectionRaycast(
      raycastEntity,
      fn,
      {
        direction: Vector3.Forward(),
        queryType: RaycastQueryType.RQT_HIT_FIRST
      }
    )

    // Simulate client-side result attachment
    raycastResultComponent.create(raycastEntity, {
      hits: [],
      direction: Vector3.Zero(),
      globalOrigin: Vector3.Zero()
    })

    await engine.update(1)
    expect(fn).toHaveBeenCalled()
```

Reply:
I get the following error if I do that change:

![Screen Shot 2023-04-19 at 14 26 01](https://user-images.githubusercontent.com/1031741/233074347-e59e4db5-2baa-419e-bb3f-f24e821b002f.png)

Reply:
MY bad, I had applied it wrong, commiting the change now